cardano/src/block/verify_chain.rs

```diff
@@ -65,17 +79,21 @@ impl ChainState {
                     return Err(Error::BlockDateInPast);
                 }
 
-                // If this is a genesis block, it should be the next
-                // epoch; otherwise it should be in the current epoch.
-                if date.get_epochid()
-                    != (last_date.get_epochid() + if date.is_boundary() { 1 } else { 0 })
-                {
-                    return Err(Error::BlockDateInFuture);
+                // New date should be this or next epoch, not further
+                if date.get_epochid() > last_date.get_epochid() {
+                    if date.get_epochid() > last_date.get_epochid() + 1 {
+                        return Err(Error::BlockDateInFuture);
+                    }
+                    // First next-epoch block supposed to be EBB,
+                    // unless we are in OBFT era
+                    if !date.is_boundary() {
+                        // TODO: validate we are in OBFT?
```

Comment:
TODO: gotta decide whether we need to add additional checks for proper OBFT switch (e.g. block version, or some other weirdo parameters from blocks). Right now we just allow new epochs to have no EBB, and that's it.